src/controllers/userController.js

```diff
@@ -106,5 +106,32 @@ const getProfile = async (request, response) => {
   });
 };
 
-const userController = { getProfile, signUp, login };
-export default userController;
+/**
+ * @description updates the profile of a user
+ * @param {object} request express request object
+ * @param {object} response express response object
+ * @param {object} next express next argument
+ * @returns {json} json
+ */
+const editProfile = async (request, response) => {
+  const { body, user } = request;
+  const updatedUser = await User.update({ ...body }, {
+    where: { id: user.id },
+    returning: true,
+    raw: true
+  });
+  if (updatedUser[0]) {
+    const {
+      password, createdAt, updatedAt, ...profile
+    } = updatedUser[1][0];
+    return responseMessage(response, 200, {
+      message: 'profile successfully updated',
+      data: { profile }
+    });
+  }
+  return responseMessage(response, 400, { error: 'the field(s) entered are invalid' });
```

Comment:
Don't you think it's better to be more specific with your response?

Reply:
Thank you for the observation, would work on a more specific response

Reply:
The response has been corrected